### ISSUE TITLE
Updating button state outside of onPressed

### ISSUE THREAD

**Issue by tylergets** [abhisheksrocks/async_button#1]
Hey there, I like the library so far. 

One thing I am having trouble figuring out how to do is update the state of the button outside the onPressed function. I want to do this so when a user submits the form it will update the button state.

In addition, it would be nice if I could trigger the onPressed the same way

Happy to help with this!

**Comment by shubhamRnaik** [comment]
good question

**Comment by abhisheksrocks** [comment]
Hey @tylergets, as you have guessed, currently there is no clean way to change button state /trigger onPressed from outside the button.

Only thing we can move outside is the code inside the onPressed function. However, that is not what you require.

**Comment by abhisheksrocks** [comment]
https://user-images.githubusercontent.com/22625661/212074548-3b538db5-05fc-4507-8f9f-5b4b37afbe91.mov

I did find a "dirty" way to trigger the onPressed from outside the button. If this suits your requirements, you can use this as a workaround for now.

Here's the gist for the same:-
https://gist.github.com/abhisheksrocks/4e9ff44e9111aa12455df1c70568ef43

Hope this helps.

**Comment by tylergets** [comment]
Thanks for the reply.

Would it be possible to initialize the AsyncButtonStateController outside the button widget and pass it in? Similiar to how TextField and TextEditingController works?

**Comment by abhisheksrocks** [comment]
I will explore this idea.

**Comment by abhisheksrocks** [comment]
Hey @tylergets! There are some news. I have pushed an update to branch [1-updating-button-state-outside-of-onpressed](https://github.com/abhisheksrocks/async_button/tree/1-updating-button-state-outside-of-onpressed) which I think should satisfy your requirements. Will push to the `main` branch soon. For now, you can check it out and let me know about any possible bugs/requirements:-

> In pubspec.yaml
```yaml
dependencies:
  flutter:
    sdk: flutter

  async_button:
    git:
      url: https://github.com/abhisheksrocks/async_button.git
      ref: 1-updating-button-state-outside-of-onpressed
```

> In your_screen.dart
```dart
class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final AsyncBtnStatesController asyncOutlinedBtnStatesController =
      AsyncBtnStatesController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            // New Way
            AsyncOutlinedBtn.withDefaultStyles(
              asyncBtnStatesController: asyncOutlinedBtnStatesController,
              onPressed: () async {
                asyncOutlinedBtnStatesController.update(AsyncBtnState.loading);
                await Future.delayed(Duration(seconds: 2));
                asyncOutlinedBtnStatesController.update(AsyncBtnState.success);
              },
              child: Text('Execute'),
            ),

            // Old Way
            AsyncOutlinedButton.withDefaultStyles(
              onPressed: (btnStateController) async {
                btnStateController.update(ButtonState.loading);
                await Future.delayed(Duration(seconds: 2));
                btnStateController.update(ButtonState.success);
              },
              child: Text('Execute'),
            ),
          ],
        ),
      ),
    );
  }

}
```

**Comment by tylergets** [comment]
Sweet! Going to check this out today!

**Comment by abhisheksrocks** [comment]
Upon some testing, I think the new implementation satisfies all your requirements. So, I have published v2.0.0-beta.1. You can now use the package as usual.
